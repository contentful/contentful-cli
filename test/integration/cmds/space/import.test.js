import nixt from 'nixt'
import { join } from 'path'
import {
  initConfig,
  // deleteSpaces,
  createSimpleSpace,
  expectedDir
} from '../../util'

const bin = join(__dirname, './../../../../', 'bin')
const org = process.env.CLI_E2E_ORG_ID

const app = () => {
  return nixt({ newlines: true, showDiff: true }).cwd(bin).base('./contentful.js ').clone()
}

var space = null
var spacesToDelete = []

beforeAll(() => {
  return initConfig()
})
beforeAll(async () => {
  space = await createSimpleSpace(org)
  spacesToDelete.push(space.sys.id)
})
// afterAll(() => {
//   // return deleteSpaces(spacesToDelete)
// })

test('should print help message', done => {
  app()
    .run('space import --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should exit 1 when no args', done => {
  app()
    .run('space import')
    .code(1)
    .expect(result => {
      const resultText = result.stderr.trim()
      expect(resultText).toMatchSnapshot('wrong response in case of no args provided')
    })
    .end(done)
})

test('should exit 1 when no space provided', done => {
  app()
    .run(`space import --content-file ${expectedDir}/export-init-space.json`)
    .code(1)
    .stderr(/Error: You need to provide a space/)
    .end((err) => {
      expect(err).toBeFalsy()
      done()
    })
})

test.skip('should import space', done => {
  app()
    .run(`space import --space-id ${space.sys.id} --content-file ${expectedDir}/export-init-space.json`)
    .stdout(/Finished importing all data/)
    .end(done)
}, 30000)
