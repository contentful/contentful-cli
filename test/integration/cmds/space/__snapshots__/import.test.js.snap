// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should exit 1 when no args: wrong response in case of no args provided 1`] = `
"Usage: contentful space import --content-file <file>

Options:
  -h, --help                 Show help                                 [boolean]
  --space-id                 ID of the destination space                [string]
  --environment-id           ID the environment in the destination space[string]
  --management-token, --mt   Contentful management API token            [string]
  --content-file             JSON file that contains data to import into a space
                                                             [string] [required]
  --content-model-only       Import only content types[boolean] [default: false]
  --skip-content-model       Skip importing content types and locales
                                                      [boolean] [default: false]
  --skip-locales             Skip importing locales   [boolean] [default: false]
  --skip-content-publishing  Skips content publishing. Creates content but does
                             not publish it           [boolean] [default: false]
  --error-log-file           Full path to the error log file            [string]
  --host                     Management API host
                                        [string] [default: \\"api.contentful.com\\"]
  --proxy                    Proxy configuration in HTTP auth format:
                             [http|https]://host:port or
                             [http|https]://user:password@host:port     [string]
  --timeout                  Timeout in milliseconds for API calls
                                                       [number] [default: 20000]
  --retry-limit              How many times to retry before an operation fails
                                                          [number] [default: 10]
  --config                   An optional configuration JSON file containing all
                             the options for a single run

Copyright 2019 Contentful
Missing required argument: content-file"
`;

exports[`should print help message: help data is incorrect 1`] = `
"Usage: contentful space import --content-file <file>

Options:
  -h, --help                 Show help                                 [boolean]
  --space-id                 ID of the destination space                [string]
  --environment-id           ID the environment in the destination space[string]
  --management-token, --mt   Contentful management API token            [string]
  --content-file             JSON file that contains data to import into a space
                                                             [string] [required]
  --content-model-only       Import only content types[boolean] [default: false]
  --skip-content-model       Skip importing content types and locales
                                                      [boolean] [default: false]
  --skip-locales             Skip importing locales   [boolean] [default: false]
  --skip-content-publishing  Skips content publishing. Creates content but does
                             not publish it           [boolean] [default: false]
  --error-log-file           Full path to the error log file            [string]
  --host                     Management API host
                                        [string] [default: \\"api.contentful.com\\"]
  --proxy                    Proxy configuration in HTTP auth format:
                             [http|https]://host:port or
                             [http|https]://user:password@host:port     [string]
  --timeout                  Timeout in milliseconds for API calls
                                                       [number] [default: 20000]
  --retry-limit              How many times to retry before an operation fails
                                                          [number] [default: 10]
  --config                   An optional configuration JSON file containing all
                             the options for a single run

Copyright 2019 Contentful"
`;
